src/jit/lowerxarch.cpp

```diff
@@ -794,6 +794,41 @@ void Lowering::LowerSIMD(GenTreeSIMD* simdNode)
         // the addr of SIMD vector with the given index.
         simdNode->gtOp1->gtFlags |= GTF_IND_REQ_ADDR_IN_REG;
     }
+    else if (simdNode->IsSIMDEqualityOrInequality())
+    {
+        LIR::Use simdUse;
+
+        if (BlockRange().TryGetUse(simdNode, &simdUse) && simdUse.User()->OperIs(GT_EQ, GT_NE) &&
```

Comment:
This code needs comments explaining what is being done. The code removed from lower.cpp was pretty well described, and I think we need similar level of detail here.